### ISSUE TITLE
Math isn't rendered properly when it ends with curly braces

### ISSUE THREAD

**Issue by jxxe** [notaapp/issues#363]
This $\LaTeX$ is valid:
```
$f(x) = \{f(x) : x \in X\}$
```
GitHub renders it correctly: $f(x) = \{f(x) : x \in X\}$

Nota, for some reason, doesn't:
<img width="83" alt="image" src="https://user-images.githubusercontent.com/64276359/217021187-3d0c5d9e-c8c4-4cd5-b448-be3e2f4eda82.png">

Nota requires it to look like this to render:
```
$f(x) = \{f(x) : x \in X\}{}$
```

It also renders if there's text directly after it:
```
$f(x) = \{f(x) : x \in X\}$abc
```

<img width="248" alt="image" src="https://user-images.githubusercontent.com/64276359/217021312-c489348d-e94e-4f92-9136-36cd5d93038d.png">

Nota has no problem if it doesn't end in a curly brace:
```
$x \in X$
```

**Comment by astoilkov** [comment]
Yeah, that's a bug in one of our dependencies. However, the dependency is open-source and no longer maintained ☹️. The fix won't be that easy.
